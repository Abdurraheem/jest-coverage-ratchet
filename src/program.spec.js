import fs from 'fs-extra'
import program from './program'
import mockCoverageReport from './coverageReport.mock'

const dir = `${process.cwd()}/__test__`
const coveragePath = `${dir}/coverage-summary.json`
const configPath = `${dir}/package.json`

beforeAll(() => {
  fs.outputJsonSync(coveragePath, mockCoverageReport)
  fs.outputJsonSync(configPath, {})
})

afterAll(() => fs.removeSync(dir))

const throwErr = (err) => {
  throw err
}

it('should not crash', () => {
  program([coveragePath, configPath]).fork(throwErr, console.log)
  expect(fs.readJsonSync(configPath)).toMatchSnapshot()
})
