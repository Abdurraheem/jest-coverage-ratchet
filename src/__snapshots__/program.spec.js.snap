// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not crash 1`] = `Object {}`;
